benchexec/tablegenerator/react-table/src/utils/filters.js

```diff
@@ -233,20 +233,34 @@ const applyMatcher = (matcher) => (data) => {
   if (!isNil(matcher.id)) {
     const { value: idValue, values: idValues } = matcher.id;
     if (idValue) {
+      // pre computing RegExp of idValue after excaping the special characters
+      let regexToCompare = new RegExp(
+        idValue.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"),
+        "ui",
+      );
```

Comment:
Can you please move the regexp construction in a separate utility function? Right now the code exists twice and it would be good to avoid this duplication.

Reply:
I have done the requested changes.